core/trino-server-main/src/main/java/io/trino/server/TrinoServer.java

```diff
@@ -13,27 +13,21 @@
  */
 package io.trino.server;
 
-import com.google.common.base.StandardSystemProperty;
-import com.google.common.primitives.Ints;
-
 import static com.google.common.base.MoreObjects.firstNonNull;
-import static com.google.common.base.Strings.nullToEmpty;
 
 public final class TrinoServer
 {
     private TrinoServer() {}
 
     public static void main(String[] args)
     {
-        String javaVersion = nullToEmpty(StandardSystemProperty.JAVA_VERSION.value());
-        String majorVersion = javaVersion.split("\\D", 2)[0];
-        Integer major = Ints.tryParse(majorVersion);
-        if (major == null || major < 22) {
+        Runtime.Version javaVersion = Runtime.version();
+        if (javaVersion.feature() < 22) {
             System.err.printf("ERROR: Trino requires Java 22+ (found %s)%n", javaVersion);
```

Comment:
Unrelated, but this was missed in the last version update

Reply:
https://github.com/trinodb/trino/pull/27288